### DISCUSSION TITLE
Themes you would like to see

### DISCUSSION THREAD

**Discussion by Accio1** [theforumhelpers/theforumhelpers.github.io#210] [Ideas]
**Edit: Please open an issue using the theme suggestion template to suggest new themes.**

Since #208 was merged, we now have the ability to implement new themes for this site! The current themes are light and dark, but it would be great to be able to have even more than that. Suggest your themes below!

**Form**
Theme Name: 

If you know which colours you want, feel free to submit all or some of them as well!
--primary: 
--secondary: 
--secondaryHover: 
--textPrimary: 
--textSecondary: 
--linkHover: 
--searchBar: 
--saturation: 

Here is a guide to what each of the colours corresponds with:
**--primary:** The main background colour.
**--secondary:** The colour of the header and footer, as well as the search button on the list of Forum Helpers page.
**--secondaryHover:** The colour of the header buttons when they are hovered over, as well as the colour the search button changes when hovered over on the list of Forum Helpers page.
**--textPrimary:** The colour used for text on the page
**--textSecondary:** The colour used for text in the header and footer
**--linkHover:** The colours that links turn when you hover over them
**--searchBar:** The background colour of the search bar
**--saturation:** The saturation of images and Ocular Statuses (if your theme is a "light" theme, you can leave saturation at 100%, if it is a darker theme, saturation values of ~60% might be better.)

@leahcimto @gosoccerboy5 @Steve0Greatness @FunctionalMetatable

**Comment by Steve0Greatness** [comment]
Terminal themes are cool:
--primary: black
--secondary: #333
--secondaryHover: grey
--textPrimary: green
--textSecondary: darkish-blue
--linkHover: yellow
--searchBar: #ccc
--saturation: 68%

**Reply by Shluffy** [reply]
what is theme name? I would run this locally and see how it goes

**Reply by Accio1** [reply]
@Shluffy, I would guess the theme name would be "Terminal".

**Reply by Steve0Greatness** [reply]
@Accio1 yes, that's the name I'm thinking of.

**Comment by leahcimto** [comment]
I'll make a scratch theme

**Reply by Steve0Greatness** [reply]
![*o*](https://en.scratch-wiki.info/w/images/thumb/Fav-it-cat-emoji.png/20px-Fav-it-cat-emoji.png)

**Reply by Accio1** [reply]
@leahcimto you could probably pull the colours right from Scratch with inspect element!

**Reply by leahcimto** [reply]
Yep!

**Reply by Shluffy** [reply]
probably what jeffalo did for oculars scratch theme

**Reply by Accio1** [reply]
@leahcimto feel free to submit a PR when you are done, and someone can add @Steve0Greatness theme above to it when you submit it.

**Comment by leahcimto** [comment]
Also, I know this has been mentioned a system option (where the theme automatically matches the user's system theme) would be cool

**Reply by Steve0Greatness** [reply]
yes, it would be cool, not sure how many people know how to do that.

**Comment by leahcimto** [comment]
Theme name: Scratch

--primary: #fcfcfc
--secondary: #4d97fe
--secondaryHover: #5686de
--textPrimary: #565e75 (this is the main text color on the homepage, black could be used instead)
--textSecondary: #ffffff
--linkHover: #4180d7
--searchBar: #ffffff
--saturation: 100%

**Reply by Shluffy** [reply]
@Accio1 could I submit a pr adding  the scratch theme and the terminal theme?

**Reply by Accio1** [reply]
@Shluffy I would wait until BarelySmooth is done working on implementing the system theme so that there aren't too many merge conflicts.

**Reply by Accio1** [reply]
@Shluffy now that #226 has been merged, feel free to go ahead and create a PR to implement these themes!

**Reply by Shluffy** [reply]
@Accio1 will get to work

**Reply by Shluffy** [reply]
i am gonna wait for #238 to be merged

**Reply by Shluffy** [reply]
how do you make the button offer the theme? see https://shluffy.github.io/tfh-fork/ and https://shluffy.github.io/tfh-fork/resources/theme.js

**Reply by Shluffy** [reply]
@BarelySmooth can you help?

**Reply by Accio1** [reply]
@Shluffy just adding it to the array in [this](https://github.com/theforumhelpers/theforumhelpers.github.io/blob/master/resources/theme.js#L28) line of theme.js will cause the button to offer the theme.

**Reply by BarelySmooth** [reply]
> how do you make the button offer the theme? see https://shluffy.github.io/tfh-fork/ and https://shluffy.github.io/tfh-fork/resources/theme.js

I don't understand. Are you asking what each variable stands for? There is [a wiki page](https://github.com/theforumhelpers/theforumhelpers.github.io/wiki/Theme-Guide) on that!

_Edit: Accio1 provided a better answer_

**Reply by Shluffy** [reply]
@Accio1 i added html.scratchMode and `,"scratch"` to the array. I'll take a closer look though. ty

**Comment by ItsEnderium** [comment]
I just want the PFPs in dark theme to be saturated again. The red in han614698's PFP was changed from #e80701 red to a11a16 tamarillo.

**Reply by Accio1** [reply]
Desaturation is by design to reduce eyestrain because, and also is recommended by [Material Design](https://material.io/design/color/dark-theme.html). Otherwise, scrolling through the list of Forum Helpers would be hard on people's eyes at night.

**Comment by ItsEnderium** [comment]
A similar version to dark, we also need a midnight/monochrome theme.

**Reply by Accio1** [reply]
Could you provide an example? I'm not sure what you mean by midnight.

**Reply by leahcimto** [reply]
Do you mean like dark but with dark blues?

**Reply by ItsEnderium** [reply]
No, a darker version of dark. Rather than grey, make the background black.

**Reply by aidan-crockett** [reply]
could you give us some specific hex or rgb colors?

**Reply by Steve0Greatness** [reply]
@gosoccerboy5, I think he's talking about something like:
```css
--primary: #000;
--secondary: #333;
--secondaryHover #222:
--textPrimary:#fff;
--textSecondary:#aaa;
--linkHover: #ccc;
--searchBar: #222;
--saturation: 0%;
--tertiary: #000;
--tertiaryHover: #111;
--textTertiary: #666;
```

**Reply by Accio1** [reply]
> No, a darker version of dark. Rather than grey, make the background black.

This is not recommended by most design standards because it can apparently be really hard on people's eyes. That is why dark grey was selected in the first place.

**Comment by CST1229** [comment]
Tried making something like an ocular theme.

Theme name: Jeffalo (or ocular)
  --primary: white;
  --secondary: #0fbd8c;
  --secondaryHover: #0e9e6a;
  --textPrimary: #575e75;
  --textSecondary: white;
  --linkHover: #4d97ff;
  --searchBar: white;
  --saturation: 100%;
  --tertiary: #0d8d59;
  --tertiaryHover: #0c7c48;
  --textTertiary: white;

**Reply by Steve0Greatness** [reply]
Looks good.
